### PR TITLE
Add options cli

### LINKED ISSUE #123: t1-linear crash files cannot be written
Hi @mdiazmel,

There is a small bug in `t1-linear` pipeline when launched with ClinicaDL (it works with Clinica).
This is due to the last line of `t1_linear_cli.py`:
```
print_crash_files_and_exit(args.logname, pipeline.base_dir)
```

There is no `logname` in `args` in the case of ClinicaDL as this argument is added at a higher level in Clinica in `cmdline.py`:
```
    parser.add_argument("-l", "--logname",
                        dest='logname',
                        default="clinica.log",
                        metavar=('file.log'),
                        help='Define the log file name (default: clinica.log)')
```

This argument (and verbose) should be added to the general command line of `clinicadl preprocessing run`.

### PR DESCRIPTION
**Pull request by mdiazmel**
Add some options to the command line:

* `--verbose`
* `--version`
* `--logname`

This PR also make a little refactor of the command line and its (sub)parsers.

Closes #123

### CODE REVIEW

File: clinicadl/clinicadl/cli.py [hunk 1]
```diff
@@ -229,6 +229,17 @@ def parse_command_line():
...
+                        default="clinicaDL.log",
+                        metavar=('file.log'),
+                        help='Define the log file name (default: clinicaDL.log)')
+    parser.add_argument("-V", "--version",
+                        dest='version',
+                        action='store_true', default=False,
+                        help="ClinicaDL's installed version")
+
     parent_parser = argparse.ArgumentParser(add_help=False)
     parent_parser.add_argument('--verbose', '-v', action='count', default=0)
```
Comment (14thibea) [inline clinicadl/clinicadl/cli.py:243]: Now that verbose is added for all parsers maybe we can remove this "parent parser"

File: clinicadl/clinicadl/cli.py [hunk 2]
```diff
@@ -239,7 +250,7 @@ def parse_command_line():
         dest='task',
         help='''****** Tasks proposed by clinicadl ******''')
 
-    subparser.required = True
+    # subparser.required = True
```
Comment (14thibea) [inline clinicadl/clinicadl/cli.py:252]: A naïve question: what is the difference when we comment this line?
I actually put this `required` argument to `True` everywhere

File: clinicadl/clinicadl/main.py [hunk 3]
```diff
@@ -9,12 +9,18 @@ def main():
...
+        import clinicadl
+        print(f"ClinicaDL version is: {clinicadl.__version__}")
+        exit(0)
     if hasattr(args, 'use_cpu'):
         if not args.use_cpu and not torch.cuda.is_available():
             raise ValueError("No GPU is available. Please add the -cpu flag to run on CPU.")
 
-    args.func(args)
+    if not args.task:
+        parser.print_help()
```
Comment (14thibea) [inline clinicadl/clinicadl/main.py:21]: I think I understood the point ^^
But maybe we can remove more of them then?

**Review by 14thibea — APPROVED** [review]
Seems ok to me!
I think we could just remove the parent parser to avoid a redundancy.

File: clinicadl/clinicadl/cli.py (hunk 1 above)
Comment (mdiazmel) [reply clinicadl/clinicadl/cli.py:243]: ~Done in 1a8e3ca~ ... Have to remove from other locations in the code...

File: clinicadl/clinicadl/cli.py (hunk 2 above)
Comment (mdiazmel) [reply clinicadl/clinicadl/cli.py:252]: The idea is to have a clear message and not an exception when launching the command line without any arguments!
In addition, it seems that required options are not advised ([see the note here](https://docs.python.org/3/library/argparse.html#required)).

File: clinicadl/clinicadl/main.py (hunk 3 above)
Comment (mdiazmel) [reply clinicadl/clinicadl/main.py:21]: Yes, at some points they can be useful (I'm thinking in some of the subparsers).

File: clinicadl/clinicadl/cli.py (hunk 1 above)
Comment (mdiazmel) [reply clinicadl/clinicadl/cli.py:243]: Well, final solution (in a1674fa) is to keep the "parent parser" for the verbose option and a general purpose parser besides (for `--logname` and `--version`).

**Comment by mdiazmel** [comment]
I'll merge this one!

**Pull request merged**